LeetCode/vDimple/String/StringPermutations_D/StringPermutations.cpp

```diff
@@ -0,0 +1,65 @@
+#include"StringPermutations.h"
+#include<map>
+
+void getStringPermutations(std::string const& str, std::vector<char>& charArr, std::vector<size_t>& charCount, size_t level, std::string& resStr, std::vector<std::string>& result)
+{
+	try
+	{
+		if (level >= str.length())
+		{
+			result.push_back(resStr);
+		}
+		else
+		{
+			for (size_t i = 0; i < charArr.size(); i++)
+			{
+				if (charCount[i] > 0)
+				{
+					resStr[level] = charArr[i];
+					++level;
```

Comment:
Instead of doing ++level and then --level, we could call the

getStringPermutations(str, charArr, charCount, level+1, resStr, result);

Reply:
Incorporated this review comment.